### PR TITLE
operator: dsa: Handle initcontainer and ConfigMap updates

### LINKED ISSUE #729: InitImage cannot be updated to DsaDevicePlugin
`UpdateDaemonSet` in dsa controller does not react to `InitImage` changes in the CRD.

### PR DESCRIPTION
**Pull request by ozhuraki**
Closes #729

### CODE REVIEW

**Review by mythi — CHANGES_REQUESTED** [review]
@ozhuraki LGTM, thanks! Please check the linter errors and let's rebase after the `go:embed` PR.

**Comment by ozhuraki** [comment]
@bartosh @mythi 
Thanks! Rebased, sorry for a  delay.

File: pkg/controllers/dsa/controller.go [hunk 1]
```diff
@@ -76,33 +76,83 @@ func (c *controller) GetTotalObjectCount(ctx context.Context, clnt client.Client
 	return len(list.Items), nil
 }
 
-func setInitContainer(spec *v1.PodSpec, imageName string) {
+func removeInitContainer(ds *apps.DaemonSet, dp *devicepluginv1.DsaDevicePlugin) {
+	newInitContainers := []v1.Container{}
+	for _, container := range ds.Spec.Template.Spec.InitContainers {
+		if container.Name == "intel-idxd-initcontainer" {
```
Comment (mythi) [inline pkg/controllers/dsa/controller.go:82]: `intel-idxd-config-initcontainer`?

**Comment by ozhuraki** [comment]
@mythi 
> intel-idxd-config-initcontainer?

Fixed, thanks!

**Review by mythi — DISMISSED** [review]

**Comment by mythi** [comment]
I missed the linter check so "conditional approval". :-)

**Comment by ozhuraki** [comment]
@mythi 
Thanks! Changes:
- Fixed linter checks

**Review by mythi — APPROVED** [review]

**Pull request merged**